File: samples/spaceshooter/scripts/state.lua
local class = require "scripts.class"

-- Base class for all game states.
local GameState = class("GameState")

-- luacheck: push ignore self game
function GameState:enter(game)
end

function GameState:update(game)
end

function GameState:draw(game)
end

function GameState:exit(game)
end
-- luacheck: pop

return GameState

File: sample/scripts/splash/state.ts
import { GameState, Game } from "../game";
import { MainMenuState } from "../mainMenu/state";

const NUMTICKS = 200;

export class SplashState implements GameState {

    private _timer = 0;

    updateBelow = false;
    drawBelow = false;

    enter(game: Game): void {
        this._timer = game.ticks + NUMTICKS;
    }
    update(game: Game): void {
        if (game.ticks > this._timer) {
            game.popState();
            game.pushState(new MainMenuState());
        }
    }
    draw(_: Game): void {
        clrs();
        font(0, 10, 10, "it'sgood! games", 2);
    }
    exit(_: Game): void {
    }
}
